examples/apps/table-detail-tea/src/main.rs

```diff
@@ -0,0 +1,211 @@
+/// A Ratatui example that demonstrates the Elm architecture with a basic list - detail
+/// application.
+///
+/// This example runs with the Ratatui library code in the branch that you are currently
+/// reading. See the [`latest`] branch for the code which works with the most recent Ratatui
+/// release.
+///
+/// [`latest`]: https://github.com/ratatui/ratatui/tree/latest
+use std::io::stdout;
+use std::time::Duration;
+use std::{fmt, panic};
+
+use crossterm::event::{
+    self, DisableMouseCapture, EnableMouseCapture, Event, KeyCode, MouseEventKind,
+};
+use crossterm::ExecutableCommand;
+use fakeit::{address, contact, name};
+use ratatui::layout::{Constraint, Layout, Rect};
+use ratatui::style::{Modifier, Style};
+use ratatui::widgets::{Block, Padding, Paragraph, Row, Table, TableState};
+use ratatui::{DefaultTerminal, Frame};
+
+#[derive(Debug)]
+struct AppModel {
+    people: Vec<Person>,
+    table_state: TableState,
+}
+
+#[derive(Debug)]
+struct Person {
+    name: String,
+    address: Address,
+    email: String,
+}
+
+#[derive(Debug)]
+struct Address {
+    street: String,
+    city: String,
+    state: String,
+    zip: String,
+}
+
+#[derive(PartialEq, Copy, Clone)]
+enum Message {
+    SelectNext,
+    SelectPrevious,
+    Quit,
+}
+
+fn main() -> color_eyre::Result<()> {
+    let terminal = ratatui::init();
+    install_panic_hook();
+    stdout().execute(EnableMouseCapture)?;
+
+    let model = AppModel::default();
+    let result = model.run(terminal);
+
+    stdout().execute(DisableMouseCapture)?;
+    ratatui::restore();
+    result
+}
+
+pub fn install_panic_hook() {
+    let original_hook = panic::take_hook();
+    panic::set_hook(Box::new(move |panic_info| {
+        stdout().execute(DisableMouseCapture).ok();
+        original_hook(panic_info);
+    }));
+}
+
+impl Default for AppModel {
+    fn default() -> Self {
+        Self {
+            people: (0..50).map(|_| Person::fake()).collect(),
+            table_state: TableState::default().with_selected(0),
+        }
+    }
+}
+
+impl AppModel {
+    fn run(mut self, mut terminal: DefaultTerminal) -> color_eyre::Result<()> {
+        loop {
+            terminal.draw(|frame| self.view(frame))?;
+
+            let mut maybe_message = handle_event()?;
+            while let Some(message) = maybe_message {
+                if message == Message::Quit {
+                    return Ok(());
+                }
+                maybe_message = self.update(message);
+            }
+        }
```

Comment:
It would be nice to add some docs about how this loop works. Mostly because it's a while loop within another loop, so from an initial read it's unclear how the interaction between the loops is intended to work.

As a reviewer seeing this for the first time (well I read it before, but that was a while ago), I don't know specifically what the intent is, I can only see that there's a nested loop. I don't have any intent to judge its correctness on.